### PR TITLE
device - Add support for Vultan Environmental Monitoring units

### PR DESCRIPTION
**Pull request by squidly**
Adding a new os for the VUTLAN-VT825 monitoring unit and a few sensors along with it.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/definitions/discovery/vutlan.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                      -
+                          oid: VUTLAN-SYSTEM-MIB::ctlUnitElementType
+                          op: '!='
+                          value: "temperature"
+            humidity:
+              data:
+                -
+                    oid: VUTLAN-SYSTEM-MIB::ctlUnitElementsTable
+                    value: VUTLAN-SYSTEM-MIB::ctlUnitElementValue
+                    #num_oid: '.1.3.6.1.4.1.318.1.1.13.3.3.1.2.2.8.{{ $index }}'
```
Comment (Jellyfrog) [inline includes/definitions/discovery/vutlan.yaml:22]: dont leave commented code

**Review by PipoCanaja — CHANGES_REQUESTED** [review]
Hi @squidly 
It seems that you added a couple of modified files from your local setup. Could you please revert them and keep only the necessary fils from your new OS ?

**Comment by squidly** [comment]
hi,
My bad, is that better ?

**Comment by mpikzink** [comment]
Hi, nice Work! What means the oDoor state? Is this clear enough for every vutlan system or can we change to a better descr.?

File: includes/vutlan.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+os: vutlan
```
Comment (PipoCanaja) [inline includes/vutlan.yml:1]: Please move this file to `includes/definitions` folder

File: mibs/vutlan/VUTLAN-SYSTEM-MIB.mib [hunk 3]
```diff
@@ -0,0 +1,3525 @@
+
+VUTLAN-SYSTEM-MIB DEFINITIONS ::= BEGIN
```
Comment (PipoCanaja) [inline mibs/vutlan/VUTLAN-SYSTEM-MIB.mib:2]: Please rename the file to `VUTLAN-SYSTEM-MIB`

**Review by PipoCanaja — CHANGES_REQUESTED** [review]

**Comment by squidly** [comment]
hi @mpikzink 
thx :) oDoor  was the way I named the door state sensor ( with value open or closed ).
I changed the desc ( and removed stalled comments ).

**Comment by PipoCanaja** [comment]
Hi @squidly 
And please add the test-data (snmprec and JSON files).
Thanx

File: tests/data/vutlan_vt825.json [hunk 4]
```diff
@@ -0,0 +1 @@
+null
```
Comment (Jellyfrog) [inline tests/data/vutlan_vt825.json:1]: This file is not correct

**Comment by squidly** [comment]
hi, I retried to produce the json data file but it still comes out empty:

```
bash-5.1# ./scripts/collect-snmp-data.php -h 192.168.64.193 -o vutlan -v vt825                                                                                                                                                    [2840/94210]
OS: vutlan                                                                                                                                                                                                                                    
Module(s): all                                                                                                                                                                                                                                
Variant: vt825                                                                                                                                                                                                                                
                                                                                                                                                                                                                                              
Capturing Data:  sysDescr.0 sysObjectID.0 SNMPv2-MIB::sysObjectID.0 SNMPv2-MIB::sysDescr.0 SNMPv2-MIB::sysName.0 SNMPv2-MIB::sysLocation.0 sysContact.0 ifDescr ifName ifAlias ifType ifOperStatus ifStackStatus entPhysicalEntry hrProcessorL
oad hrDeviceDescr hrStorageTable hrMemorySize.0 ipAdEntIfIndex ipAdEntNetMask. ipAddressIfIndex.ipv6 ipv6AddrPfxLength VUTLAN-SYSTEM-MIB::ctlUnitElementsTable entPhysicalDescr aixFsTableEntry fileSystemSpaceTable datasetTable zpoolTable d
skTable hrDeviceEntry hrProcessorEntry cdpCache lldpRemTable IP-MIB::ipNetToPhysicalPhysAddress IP-MIB::ipNetToMediaPhysAddress bgpLocalAs dot1dBasePortIfIndex dot1qVlanVersionNumber.0 diskIOEntry BRIDGE-MIB::dot1dStpProtocolSpecification
.0 dot1qTpFdbPort dot1dTpFdbPort SNMPv2-MIB::sysUpTime.0 SNMP-FRAMEWORK-MIB::snmpEngineTime.0 HOST-RESOURCES-MIB::hrSystemUptime.0 .1.3.6.1.2.1.25.3.3.1.2.196608 .1.3.6.1.2.1.25.2.3.1.6.1 .1.3.6.1.2.1.25.2.3.1.6.3 .1.3.6.1.2.1.25.2.3.1.6.
6 .1.3.6.1.2.1.25.2.3.1.6.7 .1.3.6.1.2.1.25.2.3.1.6.8 hrStorageEntry IP-MIB::icmpInMsgs.0 IP-MIB::icmpOutMsgs.0 IP-MIB::icmpInErrors.0 IP-MIB::icmpOutErrors.0 IP-MIB::icmpInEchos.0 IP-MIB::icmpOutEchos.0 IP-MIB::icmpInEchoReps.0 IP-MIB::i
cmpOutEchoReps.0 IP-MIB::icmpInDestUnreachs.0 IP-MIB::icmpOutDestUnreachs.0 IP-MIB::icmpInParmProbs.0 IP-MIB::icmpInTimeExcds.0 IP-MIB::icmpInSrcQuenchs.0 IP-MIB::icmpInRedirects.0 IP-MIB::icmpInTimestamps.0 IP-MIB::icmpInTimestampReps.0 
IP-MIB::icmpInAddrMasks.0 IP-MIB::icmpInAddrMaskReps.0 IP-MIB::icmpOutTimeExcds.0 IP-MIB::icmpOutParmProbs.0 IP-MIB::icmpOutSrcQuenchs.0 IP-MIB::icmpOutRedirects.0 IP-MIB::icmpOutTimestamps.0 IP-MIB::icmpOutTimestampReps.0 IP-MIB::icmpOut
AddrMasks.0 IP-MIB::icmpOutAddrMaskReps.0 IP-MIB::ipForwDatagrams.0 IP-MIB::ipInDelivers.0 IP-MIB::ipInReceives.0 IP-MIB::ipOutRequests.0 IP-MIB::ipInDiscards.0 IP-MIB::ipOutDiscards.0 IP-MIB::ipOutNoRoutes.0 IP-MIB::ipReasmReqds.0 IP-MIB
::ipReasmOKs.0 IP-MIB::ipReasmFails.0 IP-MIB::ipFragOKs.0 IP-MIB::ipFragFails.0 IP-MIB::ipFragCreates.0 IP-MIB::ipInUnknownProtos.0 IP-MIB::ipInHdrErrors.0 IP-MIB::ipInAddrErrors.0 IP-FORWARD-MIB::ipCidrRouteNumber.0 SNMPv2-MIB::snmpInPkt
s.0 SNMPv2-MIB::snmpOutPkts.0 SNMPv2-MIB::snmpInBadVersions.0 SNMPv2-MIB::snmpInBadCommunityNames.0 SNMPv2-MIB::snmpInBadCommunityUses.0 SNMPv2-MIB::snmpInASNParseErrs.0 SNMPv2-MIB::snmpInTooBigs.0 SNMPv2-MIB::snmpInNoSuchNames.0 SNMPv2-M
IB::snmpInBadValues.0 SNMPv2-MIB::snmpInReadOnlys.0 SNMPv2-MIB::snmpInGenErrs.0 SNMPv2-MIB::snmpInTotalReqVars.0 SNMPv2-MIB::snmpInTotalSetVars.0 SNMPv2-MIB::snmpInGetRequests.0 SNMPv2-MIB::snmpInGetNexts.0 SNMPv2-MIB::snmpInSetRequests.0
 SNMPv2-MIB::snmpInGetResponses.0 SNMPv2-MIB::snmpInTraps.0 SNMPv2-MIB::snmpOutTooBigs.0 SNMPv2-MIB::snmpOutNoSuchNames.0 SNMPv2-MIB::snmpOutBadValues.0 SNMPv2-MIB::snmpOutGenErrs.0 SNMPv2-MIB::snmpOutGetRequests.0 SNMPv2-MIB::snmpOutGetN
exts.0 SNMPv2-MIB::snmpOutSetRequests.0 SNMPv2-MIB::snmpOutGetResponses.0 SNMPv2-MIB::snmpOutTraps.0 SNMPv2-MIB::snmpSilentDrops.0 SNMPv2-MIB::snmpProxyDrops.0 UDP-MIB::udpInDatagrams.0 UDP-MIB::udpOutDatagrams.0 UDP-MIB::udpInErrors.0 UD
P-MIB::udpNoPorts.0 TCP-MIB::tcpActiveOpens.0 TCP-MIB::tcpPassiveOpens.0 TCP-MIB::tcpAttemptFails.0 TCP-MIB::tcpEstabResets.0 TCP-MIB::tcpCurrEstab.0 TCP-MIB::tcpInSegs.0 TCP-MIB::tcpOutSegs.0 TCP-MIB::tcpRetransSegs.0 TCP-MIB::tcpInErrs.
0 TCP-MIB::tcpOutRsts.0 hrSystemMaxProcesses.0 hrSystemProcesses.0 hrSystemNumUsers.0 systemStats laLoadInt.1 laLoadInt.2 laLoadInt.3 ipSystemStats ifXEntry ifEntry dot3StatsDuplexStatus dot1qPvid OSPF-MIB::ospfGeneralGroup               
Saved snmprec data /opt/librenms/tests/snmpsim/vutlan_vt825.snmprec                                                                                                                                                                           
                                                                          
```

```
bash-5.1# su - librenms -c '/opt/librenms/scripts/save-test-data.php -o vutlan -v vt825'
Starting snmpsim listening on 127.1.6.1:1161...
OS: vutlan
Module: all
Variant: vt825

Added device: 5
127.1.6.1 5 generic
Starting polling run:

Hostname:  127.1.6.1
ID:        5
OS:        generic
IP:        127.1.6.1

Enabled graphs (0):


#### Start Alerts ####
Rule #1 (testr):

Status: OK
#### End Alerts (0.0368s) ####

### Start Device Groups ###
### End Device Groups (0.0031s) ###


>>> Polled 127.1.6.1 (5) in 7.107 seconds <<<
Saved to /opt/librenms/tests/data/vutlan_vt825.json
Ready for testing!
bash-5.1# cat /opt/librenms/tests/data/vutlan_vt825.json
null

```
Any hints ?

Regards,

**Comment by murrant** [comment]
It means your device detection doesn't work.  See how it says the device is generic?

**Comment by murrant** [comment]
I tried it on my computer and it seemed to work, so must have been something wrong with your snmpsim.

**Review by murrant — DISMISSED** [review]

**Comment by Jellyfrog** [comment]
@squidly you need to fix the errors in the tests. Might help to regenerate the json test data.

**Review by PipoCanaja — APPROVED** [review]
LGTM

**Comment by PipoCanaja** [comment]
Hi @squidly 
Thanx for your contribution !

**Comment by squidly** [comment]
Glad to be of service :)

**Pull request merged**